src/mds/StrayManager.cc

```diff
@@ -637,23 +637,29 @@ void StrayManager::eval_remote_stray(CDentry *stray_dn, CDentry *remote_dn)
 {
   assert(stray_dn != NULL);
   assert(stray_dn->get_dir()->get_inode()->is_stray());
+  CDentry::linkage_t *stray_dnl = stray_dn->get_projected_linkage();
+  assert(stray_dnl->is_primary());
+  CInode *stray_in = stray_dnl->get_inode();
+  assert(stray_in->inode.nlink >= 1);
+  assert(stray_in->last == CEPH_NOSNAP);
 
   /* If no remote_dn hinted, pick one arbitrarily */
   if (remote_dn == NULL) {
-    CDentry::linkage_t *stray_dnl = stray_dn->get_projected_linkage();
-    assert(stray_dnl->is_primary());
-    CInode *stray_in = stray_dnl->get_inode();
-    assert(stray_in != NULL);
-    assert(stray_in->inode.nlink >= 1);
-
     if (!stray_in->remote_parents.empty()) {
-      remote_dn = *stray_in->remote_parents.begin();
-    } else {
-      dout(20) << __func__
-        << ": not reintegrating (no remote parents in cache)" << dendl;
+      for (compact_set<CDentry*>::iterator p = stray_in->remote_parents.begin();
+	   p != stray_in->remote_parents.end();
+	   ++p)
+	if ((*p)->last == CEPH_NOSNAP) {
```

Comment:
It makes sense to be intuitively that we wouldn't want to reintegrate into a snapshotted remote dentry, but I'm curious about what the actual problem is with doing it?  Do we just not have the infrastructure for that kind of modification?